events/bingo/webApps/bingo.js

```diff
@@ -348,16 +379,127 @@ function replaceCalledNumbers(newCalledNumbers, response) {
+        finalResponseText += "Bingo Round #" + (i + 1) + "\n";
+
+        var currentUsernameList = data[i];
+        var someoneWonAllPlayersWin = false;
+        var currentResponseText = "";
+        for (var j = 0; j < currentUsernameList.length; j++) {
+            currentResponseText += allWinnersReceive + ",";
+            currentResponseText += "Event Grant,";
+            currentResponseText += currentUsernameList[j].username;
+            currentResponseText += "\n";
+
+            if (String(currentUsernameList[j].prizeWon).toLowerCase().indexOf("all") > -1) {
+                someoneWonAllPlayersWin = true;
+            }
+        }
+
+        if (someoneWonAllPlayersWin) {
+            finalResponseText += "Amount of HFC,Type,Username\n";
+            finalResponseText += currentResponseText;
+        } else {
+            finalResponseText += "NOBODY WON 'All Players Win' in this round.";
+        }
+
+        finalResponseText += "\n\n\n";
+    }
+
+    response.statusCode = 200;
+    response.setHeader('Content-Type', 'text');
+    return response.end(finalResponseText);
+}
+
+// Handles data export request as per the user-submitted form.
+// Uses password-based authentication using the same password
+// as the db password.
+function handleDataExportRequest(formData, response) {
+    if (formData.password !== dbInfo.mySQLPassword) {
+        var responseObject = {
+            status: "error",
+            text: "Not authorized."
+        };
+    
+        response.statusCode = 401;
+        response.setHeader('Content-Type', 'application/json');
+        return response.end(JSON.stringify(responseObject));    
+    }
+
+    var query = `SHOW TABLES LIKE '${formData.prefix}_%'`;
+    connection.query(query, function(error, results, fields) {
+        if (error) {
+            var responseObject = {
+                status: "error",
+                text: "Couldn't export player data! Error while showing tables with prefix: " + error
+            };
+
+            response.statusCode = 500;
+            response.setHeader('Content-Type', 'application/json');
+            return response.end(JSON.stringify(responseObject));
+        }
+
+        if (results.length === 0) {
+            var responseObject = {
+                status: "error",
+                text: "There are no tables with that prefix! Go back and make sure you've input a valid prefix."
+            };
+
+            response.statusCode = 500;
+            response.setHeader('Content-Type', 'application/json');
+            return response.end(JSON.stringify(responseObject));
+        }
+
+        var allResults = [];
+
+        for (var i = 0; i < results.length; i++) {
+            var currentTableName;
+            for (var key in results[i]) {
+                currentTableName = results[i][key];
+            }
+            
+            query = `SELECT username,prizeWon FROM \`${currentTableName}\` WHERE username != 'BINGO BOSS'`;

```

Comment:
I think this should work if you want to do one query :
 
query = `SELECT username,prizeWon FROM \`${currentTableName}\` WHERE username != 'BINGO BOSS'`;
var unionQuery = []
for (var i = 0; i < results.length; i++) {
var currentTableName;
for (var key in results[i]) {
currentTableName = results[i][key];
unionQuery.push(
`SELECT username, prizeWon FROM `${currentTableName}`
}
unionQuery = unionQuery.join(' UNION ALL SELECT ')
var finalQuery = `
SELECT username,prizeWon FROM(${unionQuery}) as B WHERE B.username != 'BINGO BOSS'
​
Might need to vamp a little, but that basic query string worked when trying in the db

Reply:
let's say i have two of the same username in two different tables. will union make the output object contain only one of that username? or will the output object contain both?

Reply:
This just combines the tables together.   I checked manually joining 3 individual tables and the results came back to 125.  Those 3 tables had 128 listed, but the extra 3 were the bingo boss.

Reply:
fantastic! then it sounds like you're onto something. how can i then re-attribute the union'd data to their original table names?

Reply:
Also adding here in case you need it: SELECT username,prizeWon, TABLE_NAME FROM 
(SELECT username,prizeWon, 'table1' as TABLE_NAME FROM `table1` 
     UNION ALL SELECT username,prizeWon, 'table2' as TABLE_NAME FROM `table2` 
     UNION ALL SELECT username,prizeWon, 'table3' as TABLE_NAME FROM `table3`) as B WHERE B.username != 'BINGO BOSS'

Reply:
copy/paste from our Slack convo:

you can make generated columns:
```SELECT username,prizeWon, TABLE_NAME FROM 
(SELECT username,prizeWon, 'table1' as TABLE_NAME FROM `feb28_1551295337686` 
     UNION ALL SELECT username,prizeWon, 'table2' as TABLE_NAME FROM `feb28_1551394893121` 
     UNION ALL SELECT username,prizeWon, 'table3' as TABLE_NAME FROM `feb28_1551376771649`) as B WHERE B.username != 'BINGO BOSS'```

this is perfect! i'll have to revamp this code. we can probably use it as-is for tomorrow's bingo, but i'd like to fix it before merging to master.